classes/user/interest/UserInterest.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * @file classes/user/interest/UserInterest.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class UserInterest
+ *
+ * @brief UserInterest model calss
+ */
+
+namespace PKP\user\interest;
+
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use PKP\controlledVocab\ControlledVocabEntry;
+use APP\facades\Repo;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+
+class UserInterest extends Model
+{
+    use HasCamelCasing;
+
+    public const CONTROLLED_VOCAB_INTEREST = 'interest';
+    public const CONTROLLED_VOCAB_INTEREST_ASSOC_TYPE = 0;
```

Comment:
@asmecher this is fine but the reason I did not go with this approach initially because our site_id with latest implementation is set to `null` as defined by const `SITE_CONTEXT_ID` . But in `controlled_vocabs` table , columns `assoc_id` expect an `int` . so either we have to always do an int casting when storing/updating/retrieving or update the `assoc_id` to be `nullable` which kind of feel weird . What you think ?

Reply:
@As per our discussion, `controlled_vocabs` table's `assoc_id` set to nullable .